Add SRI support to Vite

renchap opened the pull request:
`VitePWA` already adds the SRI hashes to the manifest.

This PR monkey-patches `vite_ruby` to output the `integrity` attribute for JS & CSS files.

The 2nd commit is here for testing purposes only, and the last one should also be dropped before merging as its likely not the correct fix.

To test this in development, you need to not start the `vite` dev server so the manifest is generated and used.

I am a bit worried by the amount of lookups that those patches cause, especially for the stylesheets as they require a manifest scan every time.

Maybe we could cache either the tags or the lookups in non-dev mode?

I considered patching [`resolve_references`](https://github.com/ElMassimo/vite_ruby/blob/main/vite_ruby/lib/vite_ruby/manifest.rb#L154) so the `css` entries are replaced with the complete entry (like `imports` at the moment), but this will change the method return for other calls and might break them.

renchap committed:
Add patches to `vite_ruby` to output the SRI attribute

renchap committed:
Use a local cache for expensive integrity lookups

renchap committed:
Premailer no longer needs to do a lookup, we can directly load the file

renchap committed:
Do not add the big tesseract file to preload

renchap committed:
Add back vite directory to the cache

This allows the assets not to be re-compiled on next run, otherwise the `last-build-X.json` file wont be here and they will be recompiled

renchap committed:
Fix theme spec failures

The default theme file is now `application-*.css`

ClearlyClaire commented on lib/vite_ruby/sri_extensions.rb:
```diff
@@ -0,0 +1,94 @@
+  def vite_stylesheet_tag(*names, **options)
+    ''.html_safe.tap do |tags|
+      names.each do |name|
+        tags << super(name, integrity: vite_manifest.integrity_hash_for(name), **options)
```
The superclass `vite_stylesheet_tag` calls `vite_asset_path` which itself makes a lookup. Maybe it would make sense rewriting this bit to avoid doing the lookup twice (once for the path, and once for the integrity)?

renchap committed:
Remove duplicate entries from `prettierignore`

renchap committed:
Disable workbox precaching

This cached more that 4MB of assets for little benefit.

We might add this back later with a custom manifest to load less files.

renchap committed:
Add the vite binstub

With `bundle binstub vite_ruby`

renchap committed:
Do not run `yarn install` automatically when precompiling assets

renchap committed:
Avoid doing 2 lookups for a single stylesheet tag

renchap committed:
Use an instance variable for the SRI cache

renchap committed:
Vite plugins need to be in dependencies, not devDeps

renchap committed:
Remove leftovers from Jest

renchap committed:
Split vite & vitest configs, because `vitest` is only in `devDependencies`, so loading it from the config when yarn uses `--production` fails the compilation

renchap committed:
Fix wrong image import path from styles

renchap committed:
Change chunk size warning limit to fit our current setup

renchap committed:
Re-add missing Babel plugins

renchap committed:
Prevent `vite_ruby` to run yarn, always
